### PR TITLE
Chore: friday typings 5

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- fixes some types to remove `any`s and type assertions

**Why do we need this feature?**

- better typings across grafana 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by nmarrs — APPROVED** [review]
LGTM :)

File: packages/grafana-ui/src/components/Segment/SegmentInput.story.tsx [hunk 1]
```diff
@@ -6,20 +6,20 @@ import { SegmentInput, Icon, SegmentSection } from '@grafana/ui';
...
-const SegmentFrame = ({ children }: any) => (
+const SegmentFrame = ({ children }: React.PropsWithChildren) => (
   <>
     <SegmentSection label="Segment Name">{children}</SegmentSection>
   </>
 );
 
 export const BasicInput = () => {
-  const [value, setValue] = useState('some text');
+  const [value, setValue] = useState<string | number>('some text');
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Segment/SegmentInput.story.tsx:16]: When can this be a number?

**Review by ivanahuckova — APPROVED** [review]
LGTM for Loki part. Thank you for fixing this for us!! 🙇‍♀️

**Review by itsmylife — APPROVED** [review]
LGTM for graphite and influx part. 🚀

File: packages/grafana-ui/src/components/Segment/SegmentInput.story.tsx (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Segment/SegmentInput.story.tsx:16]: `SegmentInput`s `onChange` function is called with an argument` that is `string | number`. we _could_ coerce this to a string when `setValue` is called, e.g. `setValue(String(text))` and leave the state variable as a string, but 🤷

**Pull request merged**